kart/commit.py

```diff
@@ -35,13 +37,55 @@
 from .repo import KartRepoFiles
 
 
+class CommitDiffWriter(BaseDiffWriter):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if not self.spatial_filter.match_all:
+            self.RECORD_SPATIAL_FILTER_STATS = True
+            self.spatial_filter_pk_conflicts = {
+                ds_path: [] for ds_path in self.all_ds_paths
+            }
+            self.remove_from_wc_post_commit = {
+                ds_path: [] for ds_path in self.all_ds_paths
+            }
+        else:
+            self.RECORD_SPATIAL_FILTER_STATS = False
+            self.spatial_filter_pk_conflicts = self.remove_from_wc_post_commit = None
```

Comment:
minor: IMO avoid `a = b = c` lines; it's always clearer to just put them on two lines

Reply:
Done